config/config.go

```diff
@@ -400,6 +375,10 @@ func validateConfig(cfg *Config) {
 	fixDeprecatedLog(cfg)
 
 	fixDeprecatedPorts(cfg)
+
+	fixDeprecatedRedis(cfg)
+
+	return validateRedisConfig(cfg)
```

Comment:
These would be nicer as `RedisConfig` methods IMO.

Maybe we can also merge them into just `validate`?  
Since anyways we're already "fixing" inside `validateConfig`, I don't think it won't be out of place.

And if we merge those functions we can also return an error if both a new and old options have values, instead of ignoring the old value as the current `fixDeprecatedRedis` does.

Reply:
implemented it as `RedisConfig.validateConfig`